From 3cdc1d0654dcd8e6eb3fc01d95fcccafe68cbb47 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Thu, 12 Oct 2017 18:41:58 +0200
Subject: Fix (my) yakuake on a wayland session

KWindowSystem::activeWindow and winId are two entirely different things.

Reviewed by Eike
---
 app/mainwindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/mainwindow.cpp b/app/mainwindow.cpp
index 6fa9160..028bb9c 100644
--- a/app/mainwindow.cpp
+++ b/app/mainwindow.cpp
@@ -1023,7 +1023,7 @@ void MainWindow::wmActiveWindowChanged()
         return;
     }
 
-    if (!Settings::keepOpen() && isVisible() && KWindowSystem::activeWindow() != winId()) {
+    if (!Settings::keepOpen() && hasFocus()) {
         toggleWindowState();
     }
 }
-- 
cgit v0.11.2

